dbt/adapters/databricks/connections.py

```diff
@@ -47,15 +53,22 @@ def __pre_deserialize__(cls, data: Dict[Any, Any]) -> Dict[Any, Any]:
         return data
 
     def __post_init__(self) -> None:
-        # spark classifies database and schema as the same thing
-        if self.database is not None and self.database != self.schema:
+        session_properties = self.session_properties or {}
+        if CATALOG_KEY_IN_SESSION_PROPERTIES in session_properties:
+            if self.database is None:
+                self.database = session_properties[CATALOG_KEY_IN_SESSION_PROPERTIES]
+                del session_properties[CATALOG_KEY_IN_SESSION_PROPERTIES]
+            else:
+                raise dbt.exceptions.RuntimeException(
+                    f"`catalog` or `database` and `{CATALOG_KEY_IN_SESSION_PROPERTIES}` "
```

Comment:
Maybe we can add a test case for this? 

And what if I have a config like this:
```
{
  "database": "db",
  "catalog": "ct",
}
```
Will it throw exceptions?

Reply:
Yes, it throws an exception. `Got duplicate keys: (catalog) all map to "database"`.
Added the test and modified the error message here to follow it.